src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/inplace_resolve_io.cpp

```diff
@@ -0,0 +1,333 @@
+
+        ov::pass::Serialize serializer("Subgraph0.xml", "Subgraph0.bin");
+        serializer.run_on_model(function);
+    }
+
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+        for (size_t i = 0lu; i < funcInputs.size(); i++) {
+            const auto& funcInput = funcInputs[i];
+            ov::Tensor tensor;
+            if (i == 2) {  // sequence_lengths
+                tensor = ov::Tensor{ov::element::i32, targetInputStaticShapes[i]};
+                auto data = tensor.data<ov::element_type_traits<ov::element::i32>::value_type>();
+                for (size_t i = 0lu; i < ov::shape_size(targetInputStaticShapes[i]); i++) {
+                    data[i] = 10;
+                }
+            } else {
+                if (funcInput.get_element_type().is_real()) {
+                    ov::test::utils::InputGenerateData in_data;
+                    in_data.start_from = 0;
+                    in_data.range = 10;
+                    in_data.resolution = 1000;
+                    tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], in_data);
+                } else {
+                    tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(),
+                                                                     targetInputStaticShapes[i]);
+                }
+            }
+            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
+        }
+    }
+};
+
+TEST_P(RNNConcatSubgraphTest, smoke_CompareWithRefs) {
+    run();
+}
+
+namespace {
+
+const std::vector<std::vector<InputShape>> inputShapes2 = {
+    {
+        // static
+        {{2, 10, 8}, {{2, 10, 8}}}, // X [batch_size, seq_length, input_size]
+        {{2, 1, 3}, {{2, 1, 3}}},   // H_t [batch_size, num_directions, hidden_size]
+        {{2}, {{2}}}                // [batch_size]
+    },
+    {
+        // dynamic
+        {{-1, 10, 8}, {{2, 10, 8}}}, // param0
+        {{-1, 1, 3}, {{2, 1, 3}}}, // param1
+        {{-1}, {{2}}},  // param2
+    },
+};
+
+INSTANTIATE_TEST_SUITE_P(smoke_inplace_resolve_io, RNNConcatSubgraphTest,
+                        ::testing::ValuesIn(inputShapes2),
+                        RNNConcatSubgraphTest::getTestCaseName);
+}  // namespace
+
+class SoftmaxAddReshapeOutputSubgraphTest : public InplaceResolveIOTestBase {
+/*This test runs the following subgraph:
+
+                      param
+                        |
+                        |
+                      Softmax
+                     /     \
+                    /       \
+                   Add     Reshape0
+                    |         |
+                    |         |
+                 Result0   Result1
+
+expect edge Reshape1->Result1 to be referenced by its upstreams, instead of referencing to its upstreams.
```

Comment:
The comment is not very clear, could you please rephrase it?

Reply:
done